schemas/post-artifact-request.yml

```diff
@@ -3,6 +3,110 @@ title:              "Post Artifact Request"
 description: "Request a authorization to put and artifact or posting of a URL as an artifact. Note that the `storageType` property is referenced in the response as well."
 type:               object
 oneOf:
+  - title: "Blob Artifact Request"
+    description: |
+      Request a list of requests in a generalized format which can be run to
+      upload an artifact to storage managed by the queue.
+    type: object
+    properties:
+      storageType:
+        description: Artifact storage type, in this case `'blob'`
+        type: string
+        enum:
+          - blob
+      expires:
+        description: |
+          Date-time after which the artifact should be deleted. Note, that
+          these will be collected over time, and artifacts may remain
+          available after expiration. S3 based artifacts are identified in
+          azure table storage and explicitly deleted on S3 after expiration.
+        type: string
+        format: date-time
+      contentType:
+        description: |
+          Artifact mime-type, when uploading artifact to the signed
+          `PUT` URL returned from this request this must given with the
+           `ContentType` header. Please, provide correct mime-type,
+           this make tooling a lot easier, specifically,
+           always using `application/json` for JSON artifacts.
+        type: string
+        maxLength: 255
+      encoding:
```

Comment:
Okay, so I spent a few more thought cycles on this... And talked to bstack.

The natural thing is to have `x-amz-meta-content-sha256` be the hash of the contents in it's canonical form, that is before it is being gzipped.
 * This is what the now obsolute `content-md5` header specifies
 * If making the request through XHR in the browser, the browser will automatically gzip decode, so even if you can access the header you won't be able to verify the hash.

But then we actually do need two different hashes:
 * The canonical hash for `x-amz-meta-content-sha256` (that hash of uncompressed resource)
 * The object hash, or part hash, for `x-amz-content-sha256` to ensure correct transfer.

I propose we do this by changing the data format to be:
```js
{
  contentEncoding: '...', // I propose we limit this to 'gzip', 'br' or 'identity'
  contentType: '...',
  contentSha256: '...', // hash of the entire resource before gzipping
  contentLength: 45345,
  parts: [
    {
      sha256: '...', // 
      size: 345435,
    },
    // One or more parts, but always at-least one.
    // Because we have no use for sha256 of the entire artifact without content-encoding
    // unless it is being uploaded in a single request.
  ],
}
```